pkg/etcd/etcd.go

```diff
@@ -547,10 +547,24 @@ func (c *CDCEtcdClientImpl) PutCaptureInfo(
 	return cerror.WrapError(cerror.ErrPDEtcdAPIError, err)
 }
 
-// DeleteCaptureInfo delete capture info from etcd.
-func (c *CDCEtcdClientImpl) DeleteCaptureInfo(ctx context.Context, id string) error {
-	key := GetEtcdKeyCaptureInfo(c.ClusterID, id)
+// DeleteCaptureInfo delete all capture related info from etcd.
+func (c *CDCEtcdClientImpl) DeleteCaptureInfo(ctx context.Context, captureID string) error {
+	key := GetEtcdKeyCaptureInfo(c.ClusterID, captureID)
 	_, err := c.Client.Delete(ctx, key)
+	if err != nil {
+		return cerror.WrapError(cerror.ErrPDEtcdAPIError, err)
+	}
+	// we need to clean all task position related to this capture when the capture is offline
+	// otherwise the task positions may leak
+	// TODO: use changefeed's name space
+	taskKey := TaskPositionKeyPrefix(c.ClusterID, model.DefaultNamespace)
+	// the taskKey format is /tidb/cdc/{clusterID}/{namespace}/task/position/{captureID}
+	taskKey = fmt.Sprintf("%s/%s", taskKey, captureID)
+	_, err = c.Client.Delete(ctx, taskKey, clientv3.WithPrefix())
+	if err != nil {
+		log.Warn("delete task position failed",
```

Comment:
Should we print other info? like cluster id and capture id.

Reply:
good idea.